### PR TITLE
Use math.Order for comparing bigints instead of i8

### PR DESCRIPTION
**Pull request by joachimschmidt557**
Implements Proposal https://github.com/ziglang/zig/issues/4781

### CODE REVIEW

File: lib/std/math/big/int.zig [hunk 1]
```diff
@@ -535,12 +535,12 @@ pub const Int = struct {
     }
 
     /// Returns -1, 0, 1 if |a| < |b|, |a| == |b| or |a| > |b| respectively.
```
Comment (LemonBoy) [inline lib/std/math/big/int.zig]: This doc comment is now outdated

File: lib/std/math/big/int.zig [hunk 2]
```diff
@@ -551,21 +551,25 @@ pub const Int = struct {
...
         } else if (a.limbs[i] > b.limbs[i]) {
-            return 1;
+            return .gt;
         } else {
-            return 0;
+            return .eq;
         }
     }
 
     /// Returns -1, 0, 1 if a < b, a == b or a > b respectively.
```
Comment (LemonBoy) [inline lib/std/math/big/int.zig]: Ditto

File: lib/std/math/big/rational.zig [hunk 3]
```diff
@@ -327,17 +327,17 @@ pub const Rational = struct {
     }
 
     /// Returns -1, 0, 1 if a < b, a == b or a > b respectively.
```
Comment (LemonBoy) [inline lib/std/math/big/rational.zig]: Ditto

File: lib/std/math/big/rational.zig [hunk 4]
```diff
@@ -327,17 +327,17 @@ pub const Rational = struct {
...
-    pub fn cmp(a: Rational, b: Rational) !i8 {
+    pub fn cmp(a: Rational, b: Rational) !math.Order {
         return cmpInternal(a, b, true);
     }
 
     /// Returns -1, 0, 1 if |a| < |b|, |a| == |b| or |a| > |b| respectively.
```
Comment (LemonBoy) [inline lib/std/math/big/rational.zig]: Ditto.

File: lib/std/math/big/int.zig [hunk 5]
```diff
@@ -534,7 +534,8 @@ pub const Int = struct {
         return out_stream.writeAll(str);
     }
 
-    /// Returns -1, 0, 1 if |a| < |b|, |a| == |b| or |a| > |b| respectively.
+    /// Returns math.Order.lt, math.Oder.eq, math.Order.gt if |a| < |b|, |a| ==
```
Comment (LemonBoy) [inline lib/std/math/big/int.zig]: Typo, `Oder` -> `Order`

Comment (joachimschmidt557) [reply lib/std/math/big/int.zig]: Oops, will fix that in all of the doc comments

**Pull request merged**